addons/godot-next/objects/message_dispatcher.gd

```diff
@@ -0,0 +1,51 @@
+# MessageDispatcher
+# author: MunWolf (Rikhardur Bjarni Einarsson)
+# license: MIT
+# copyright: Copyright (c) 2019 Rikhardur Bjarni Einarsson
+# brief_description: A message handler for non predefined signals, can be extended by any class
+#                    Node, Reference, Resource etc, can also be directly attached to a Node.
+extends Object
+
+class_name MessageDispatcher
+
+##### PROPERTIES #####
+
+var _message_handlers := {}
+
+##### PUBLIC METHODS #####
+
+# Connect a handler, obj has to have a function that corresponds to the parameter.
+#    message_type: type of the message, we call obj.function(message) based on this.
+#    obj: object that holds the callback.
+#    function: function to be called on the object.
+func connect_message(message_type: String, obj: Object, function: String) -> void:
```

Comment:
You could (if you wanted to) also convert all of the Object + method string combos as just a FuncRef object, using the `call_func()` method on it. So, rather than `handler[0].call(handler[1]...)`, you'd have `func.call_func(message_type, message_data)`. Keeping it separated is useful if you have the ability to individually edit the object or method used (if they were properties), but since you storing it as an entire record, a FuncRef may be more clear of your intent than an Array.

Reply:
I don't mind doing this, but that would mean we give up any checking if its valid as currently FuncRef does not allow you to check its internals and has no method for such checking.

Reply:
If https://github.com/godotengine/godot/pull/28525 gets merged we can easily go for this.